internal/repository/verify.go

```diff
@@ -4,22 +4,36 @@ package repository
 
 import (
 	"context"
+	"errors"
 
 	"github.com/gittuf/gittuf/internal/gitinterface"
 	"github.com/gittuf/gittuf/internal/policy"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing"
 )
 
+var ErrRefStateDoesNotMatchRSL = errors.New("Git reference's current state does not match latest RSL entry") //nolint:stylecheck
```

Comment:
Add godoc explaining what this means / how a repo can get into this state / how to resolve?